triton-vm/src/table/base_matrix.rs

```diff
@@ -46,17 +46,108 @@ pub struct BaseMatrices {
+            &jump_stack_matrix,
+        );
+
         Self {
-            program_matrix: Self::derive_program_matrix(program),
-            instruction_matrix: Self::derive_instruction_matrix(&aet, program),
-            op_stack_matrix: Self::derive_op_stack_matrix(&aet),
-            ram_matrix: Self::derive_ram_matrix(&aet),
-            jump_stack_matrix: Self::derive_jump_stack_matrix(&aet),
-            processor_matrix: aet.processor_matrix,
+            program_matrix,
+            instruction_matrix,
+            processor_matrix,
+            op_stack_matrix,
+            ram_matrix,
+            jump_stack_matrix,
             hash_matrix: aet.hash_matrix,
         }
     }
 
+    fn add_clock_jump_differences_to_processor_matrix(
+        mut processor_matrix: Vec<[BFieldElement; processor_table::BASE_WIDTH]>,
+        op_stack_matrix: &[[BFieldElement; op_stack_table::BASE_WIDTH]],
+        ram_matrix: &[[BFieldElement; ram_table::BASE_WIDTH]],
+        jump_stack_matrix: &[[BFieldElement; jump_stack_table::BASE_WIDTH]],
+    ) -> Vec<[BFieldElement; processor_table::BASE_WIDTH]> {
+        let one = BFieldElement::one();
+
+        // get clock jump differences for all 3 memory-like tables
+        let op_stack_mp_column = usize::from(OpStackBaseTableColumn::OSP);
+        let op_stack_clk_column = usize::from(OpStackBaseTableColumn::CLK);
+        let mut all_clk_jump_differences = vec![];
+        for (curr_row, next_row) in op_stack_matrix.iter().tuple_windows() {
+            if next_row[op_stack_mp_column] == curr_row[op_stack_mp_column] {
+                let clock_jump_diff = next_row[op_stack_clk_column] - curr_row[op_stack_clk_column];
+                if clock_jump_diff != one {
+                    all_clk_jump_differences.push(clock_jump_diff);
+                }
+            }
+        }
+
+        let ramp_mp_column = usize::from(RamBaseTableColumn::RAMP);
+        let ram_clk_column = usize::from(RamBaseTableColumn::CLK);
+        for (curr_row, next_row) in ram_matrix.iter().tuple_windows() {
+            if next_row[ramp_mp_column] == curr_row[ramp_mp_column] {
+                let clock_jump_diff = next_row[ram_clk_column] - curr_row[ram_clk_column];
+                if clock_jump_diff != one {
+                    all_clk_jump_differences.push(clock_jump_diff);
+                }
+            }
+        }
+
+        let jump_stack_mp_column = usize::from(JumpStackBaseTableColumn::JSP);
+        let jump_stack_clk_column = usize::from(JumpStackBaseTableColumn::CLK);
+        for (curr_row, next_row) in jump_stack_matrix.iter().tuple_windows() {
+            if next_row[jump_stack_mp_column] == curr_row[jump_stack_mp_column] {
+                let clock_jump_diff =
+                    next_row[jump_stack_clk_column] - curr_row[jump_stack_clk_column];
+                if clock_jump_diff != one {
+                    all_clk_jump_differences.push(clock_jump_diff);
+                }
+            }
+        }
```

Comment:
@sshine, do you happen to have an idea on how to de-duplicate this? The issue seems to be that the three different matrices, which are Vec<[BFieldElement; respective_table::BASE_WIDTH]> have different base widths.